pkg/gofr/datasource/opentsdb/annotation.go

```diff
@@ -0,0 +1,329 @@
+	StatusCode int
+
+	// UpdateAnnotations holds the list of annotations involved in a bulk update.
+	UpdateAnnotations []Annotation `json:"InvolvedAnnotations,omitempty"`
+
+	// ErrorInfo contains details about any errors that occurred during the bulk operation.
+	ErrorInfo map[string]interface{} `json:"error,omitempty"`
+
+	// TSUIDs holds the list of TsUIDs for annotations that should be deleted.
+	// If empty or nil, the global flag is used.
+	TSUIDs []string `json:"tsuids,omitempty"`
+
+	// StartTime is the Unix epoch timestamp for the start of the deletion request.
+	StartTime int64 `json:"startTime,omitempty"`
+
+	// EndTime is the optional Unix epoch timestamp for when the events ended.
+	EndTime int64 `json:"endTime,omitempty"`
+
+	// Global indicates whether global annotations should be deleted for the given time range.
+	Global bool `json:"global,omitempty"`
+
+	// TotalDeleted holds the total number of annotations successfully deleted in the bulk operation.
+	TotalDeleted int64 `json:"totalDeleted,omitempty"`
+
+	logger Logger
+	tracer trace.Tracer
+	ctx    context.Context
+}
+
+// BulkAnnotationDeleteInfo holds the parameters for a bulk annotation delete operation.
+type BulkAnnotationDeleteInfo struct {
+	// TsUIDs holds the list of TSUIDs for annotations that should be deleted.
+	// If empty or nil, the global flag is used.
+	TSUIDs []string `json:"tsuids,omitempty"`
+
+	// StartTime is the Unix epoch timestamp for the start of the deletion request.
+	StartTime int64 `json:"startTime,omitempty"`
+
+	// EndTime is the optional Unix epoch timestamp for when the events ended.
+	EndTime int64 `json:"endTime,omitempty"`
+
+	// Global indicates whether global annotations should be deleted for the given time range.
+	Global bool `json:"global,omitempty"`
+}
+
+// BulkDeleteResp contains the results of a bulk annotation delete operation.
+type BulkDeleteResp struct {
+	BulkAnnotationDeleteInfo
+}
+
+func (bulkAnnotResp *BulkAnnotationResponse) SetStatus(code int) {
+	setStatus(bulkAnnotResp.ctx, bulkAnnotResp, code, "SetStatus-BulkAnnotation", bulkAnnotResp.logger)
+}
+
+func (bulkAnnotResp *BulkAnnotationResponse) setStatusCode(code int) {
+	bulkAnnotResp.StatusCode = code
+}
+
+func (bulkAnnotResp *BulkAnnotationResponse) GetCustomParser() func(respCnt []byte) error {
+	return getCustomParser(bulkAnnotResp.ctx, bulkAnnotResp, "GetCustomParser-BulkAnnotation", bulkAnnotResp.logger,
+		func(resp []byte) error {
+			originContents := string(resp)
+			tag := strings.Split(originContents, " ")[0]
+
+			var resultBytes []byte
+
+			if strings.Contains(tag, "error") || strings.Contains(tag, "totalDeleted") {
+				resultBytes = resp
+			} else if strings.Contains(tag, "tsuid") {
+				resultBytes = []byte(fmt.Sprintf(`{"InvolvedAnnotations":%s}`, originContents))
+			} else {
+				return fmt.Errorf("unrecognized bulk annotation response: %s", originContents)
+			}
+
+			return json.Unmarshal(resultBytes, &bulkAnnotResp)
```

Comment:
This way to parse is an hack.

I would recommend using something better than this.

I will try to suggest you something when I will have a computer. Only a phone for now

Are these covered by a unit tests?